docker_test.go

```diff
@@ -83,3 +83,33 @@ func TestPortMappingLabels(t *testing.T) {
 	assert.NotNil(t, pm.Labels)
 	assert.Equal(t, 0, len(*pm.Labels))
 }
+
+func TestDockerPersistentVolume(t *testing.T) {
+	docker := NewDockerApplication()
+
+	pvol := &PersistentVolume{Size: 2048}
+	pvol.SetType("root")
+	pvol.AddConstraint("bat")
+
+	assert.Equal(t, 1, len(*pvol.Constraints))
+	assert.Equal(t, []string{"bat"}, (*pvol.Constraints)[0])
+	pvol.EmptyConstraints()
+	assert.NotNil(t, pvol.Constraints)
+	assert.Equal(t, 0, len(*pvol.Constraints))
+
+	vol := Volume{
+		ContainerPath: "/foo",
+		HostPath:      "/bar",
+		Mode:          "RW",
+		Persistent:    pvol,
+	}
+
+	docker.Container.AddVolume(vol)
+
+	assert.Equal(t, 1, len(*docker.Container.Volumes))
+	assert.Equal(t, "/foo", (*docker.Container.Volumes)[0].ContainerPath)
```

Comment:
Let's extract the first volume into a variable here and run our assertions against it.